src/dune_lang/syntax.mli

```diff
@@ -58,18 +58,19 @@ module Warning : sig
 end
 
 (** [create ~name ~desc supported_versions] defines a new syntax.
-    [supported_version] is the list of the last minor version of each supported
-    major version. [desc] is used to describe what this syntax represent in
-    error messages. *)
-val create : name:string -> desc:string -> Version.t list -> t
+    [supported_version] is the list of all the supported versions paired with
+    the versions of the dune lang in which they where introduced. [desc] is used
+    to describe what this syntax represent in error messages. *)
+val create :
+  name:string -> desc:string -> (Version.t * [ `Since of Version.t ]) list -> t
 
 (** Return the name of the syntax. *)
 val name : t -> string
 
 (** Check that the given version is supported and raise otherwise. *)
-val check_supported : t -> Loc.t * Version.t -> unit
+val check_supported : lang_ver:Version.t -> t -> Loc.t * Version.t -> unit
```

Comment:
```suggestion
val check_supported : dune_lang_version:Version.t -> t -> Loc.t * Version.t -> unit
```

Just to make things fully explicit. This is not something we use all over the place, so we should prioritise explicitness over conciseness.